src/lv_message_efficient.cc

```diff
@@ -117,82 +110,78 @@ namespace grpc_labview
 
     //---------------------------------------------------------------------
     //---------------------------------------------------------------------
-    const char *LVMessageEfficient::ParseBytes(google::protobuf::uint32 tag, const MessageElementMetadata& fieldInfo, uint32_t index, const char *ptr, ParseContext *ctx)
+    const char* LVMessageEfficient::ParseBytes(google::protobuf::uint32 tag, const MessageElementMetadata& fieldInfo, uint32_t index, const char* ptr, ParseContext* ctx)
     {
-        return ParseString(tag, fieldInfo, index, ptr, ctx);    
+        return ParseString(tag, fieldInfo, index, ptr, ctx);
     }
 
     //---------------------------------------------------------------------
     //---------------------------------------------------------------------
-    const char *LVMessageEfficient::ParseNestedMessage(google::protobuf::uint32 tag, const MessageElementMetadata& fieldInfo, uint32_t index, const char *protobuf_ptr, ParseContext *ctx)
-    {    
+    const char* LVMessageEfficient::ParseNestedMessage(google::protobuf::uint32 tag, const MessageElementMetadata& fieldInfo, uint32_t index, const char* protobuf_ptr, ParseContext* ctx)
+    {
         auto metadata = fieldInfo._owner->FindMetadata(fieldInfo.embeddedMessageName);
-        
+
         LVMessageEfficient nestedMessage(metadata);
 
-        if (fieldInfo.isRepeated) {
+        if (fieldInfo.isRepeated)
+        {
             // if the array is not big enough, resize it to 2x the size
             auto numElements = 128;
             auto elementIndex = 0;
             auto clusterSize = metadata->clusterSize;
-            auto alignment = metadata->alignmentRequirement;
             auto arraySize = numElements * clusterSize;
             char _fillData = '\0';
 
             // Get the _repeatedMessageValues vector from the map
-            auto _repeatedMessageValuesIt = _repeatedMessageValuesMap.find(metadata->messageName);
+            auto _repeatedMessageValuesIt = _repeatedMessageValuesMap.find(fieldInfo.fieldName);
```

Comment:
NOTE: This effectively fixes #378 where a message has two repeated fields of the same message type. Since we were tracking the data via the message name rather than the field name, the second field was adding its data to the end of the first field.